dist/jso.d.ts

```diff
@@ -0,0 +1,145 @@
+export declare type ResponseType =
+  'id_token token' |
+  'code id_token'  |
+  'token'          |
+  'code'
+
+export declare type TokenType =
+  'Bearer'
+
+export declare interface Request {
+  providerID: string,
+  response_type: ResponseType
+  redirect_uri?: string,
+  client_id: string,
+  state: string
+  openid: boolean,
+  scope?: string,
+  nonce?: string
+  scopes?: string[],
+  restoreHash?: string
+}
+
+export declare interface AuthorizationResponse extends Response {
+  code: string
+}
+
+export declare interface ImplicitResponse extends Response {
+  access_token: string
+}
+
+export declare interface Response {
+  expires_in: number
+  scope: string
+  state: string
+  token_type: TokenType
+}
+
+export declare interface Token {
+  access_token: string
+  expires: number
+  expires_in: number
+  received: number
+  scopes: string[]
+  state: string
+  token_type: TokenType
+}
+
+export declare class JSO extends EventEmitter {
+  constructor(config: ConfigOptions)
+
+  callback(data?: Response | string): Token | Promise<Token> | undefined
+  checkToken(opts?: TokenOptions): Token | undefined
+  configure(config: ConfigOptions): void
+  dump(): Token[]
+  getProviderID(): string
+  getToken(opts?: TokenOptions): Promise<Token>
+  setLoader(loader: BasicLoader): void
+  setStore(newstore: Store): void
+  wipeTokens(): void
+}
+
+export declare class EventEmitter {
+  on(type: string, callback: (args: any) => void): void
+  emit(type: string, ...args: any[]): void
+}
+
+export declare interface TokenOptions {
+  scopes?: ScopeOptions,
+  allowredir?: boolean
+}
+
+export declare class Config {
+  constructor(...opts: ConfigOptions[])
+  
+  has(key: string): Boolean
+  
+  getValue<K extends keyof ConfigOptions>(
+    key: K,
+    defaultValue: ConfigOptions[K],
+    isRequired: Boolean
+  ): ConfigOptions[K]
+}
+
+export declare interface ConfigOptions {
+  providerID?: string
+  client_id: string
+  client_secret?: string
+  authorization: string
+  token: string
```

Comment:
`token` doesn't seem to be a valid option for the constructor?

https://github.com/andreassolberg/jso#initializing-the-client

Reply:
Apparently it's the endpoint URL for exchanging a code for an access token:
https://github.com/andreassolberg/jso/issues/89

Reply:
It is at least optional, though.